### DISCUSSION TITLE
How does CSS class tree shaking work? (It's non-deterministic for me)

### DISCUSSION THREAD

**Discussion by jtlapp** [tailwindlabs/tailwindcss#11605] [Help] [answered]
I'm having trouble with my builds. Sometimes CSS classes make it into the build, sometimes they don't. It's always the result of a change I make to the code, but it's not consistent for those changes.

To help me troubleshoot this, could someone point me to a description of how Tailwind does tree shaking? How can I determine whether a class will appear in the build or not?

I'm using SvelteKit, if that matters, and the classes are dynamically added.

Thanks!

**Comment by jtlapp** [comment]
I solved my problem. I'm working in a monorepo pulling components from one workspace into another. [Here is the solution](https://dev.to/krzysztofzuraw/compiling-tailwind-css-components-in-monorepo-1ljk).

I'd still love to read about how tree shaking works, though.

**Reply by mybearworld** [reply, marked as answer]
[Here](https://tailwindcss.com/docs/content-configuration#class-detection-in-depth)'s the page of the docs.
